src/DebuggerData.h

```diff
@@ -19,58 +19,56 @@ struct MemoryLayout
 	int mapperSize[4][4];
 };
 
+struct Breakpoint {
+	enum Type { BREAKPOINT, WATCHPOINT_MEMREAD, WATCHPOINT_MEMWRITE,
+	            WATCHPOINT_IOREAD, WATCHPOINT_IOWRITE, CONDITION };
+	Type type;
+	QString id;
+	quint16 address;
+	// end for watchpoint region
+	quint16 regionEnd;
+	// gui specific condition variables
+	qint8 ps;
+	qint8 ss;
+	qint16 segment;
+	// general condition
+	QString condition;
+	// compare content
+	bool operator==(const Breakpoint &bp) const;
+};
+
 class Breakpoints
 {
 public:
-	Breakpoints();
-
-	enum Type { BREAKPOINT = 0, WATCHPOINT_MEMREAD, WATCHPOINT_MEMWRITE,
-	            WATCHPOINT_IOREAD, WATCHPOINT_IOWRITE, CONDITION };
-
 	void clear();
 
 	void setMemoryLayout(MemoryLayout* ml);
 	void setBreakpoints(const QString& str);
 	QString mergeBreakpoints(const QString& str);
 	int breakpointCount();
-	bool isBreakpoint(quint16 addr, QString *id = nullptr);
-	bool isWatchpoint(quint16 addr, QString *id = nullptr);
+	bool isBreakpoint(quint16 addr, QString *id = nullptr, bool checkSlot = true);
+	bool isWatchpoint(quint16 addr, QString *id = nullptr, bool checkSlot = true);
 
 	/* xml session file functions */
 	void saveBreakpoints(QXmlStreamWriter& xml);
 	void loadBreakpoints(QXmlStreamReader& xml);
 
 	int findBreakpoint(quint16 addr);
-	int findNextBreakpoint();
 
-	static QString createSetCommand(Type type, int address,
+	static QString createSetCommand(Breakpoint::Type type, quint16 address,
 	                                qint8 ps = -1, qint8 ss = -1, qint16 segment = -1,
-	                                int endRange = -1, QString condition = QString());
+	                                quint16 endRange = -1, QString condition = QString());
```

Comment:
Hmm, not sure, but I think `quint16 endRange = -1` is wrong.
Internally in `createSetCommand()` we test `endRange > address`. And from `DebuggerForm::toggleBreakpoint()` we call `createSetCommand()` with only 5 parameters (so with `endRange = -1`. I think that combination wrongly concludes there's a big range (till 0xffff), while before it was a single address.

Reply:
Yes, you are right. The ideal solution would be the default value endRange = address, but that is not possible. But, we just need to use a endRange smaller than address and createSetCommand will ignore it. So I guess `qint16` is OK.